src/web/gwc/src/main/java/org/geoserver/gwc/web/blob/BlobStoreTypeStore.java

```diff
@@ -0,0 +1,133 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.gwc.web.blob;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Comparator;
+
+import org.apache.wicket.markup.html.panel.Panel;
+import org.apache.wicket.model.IModel;
+import org.geowebcache.config.BlobStoreConfig;
+import org.geowebcache.config.FileBlobStoreConfig;
+import org.geowebcache.s3.S3BlobStoreConfig;
+
+/**
+ *
+ * A singleton store of all supported blobstore types.
+ *
+ * @author Niels Charlier
+ */
+public class BlobStoreTypeStore {
+
+    protected static BlobStoreTypeStore THE_STORE = new BlobStoreTypeStore();
+
+    public static BlobStoreTypeStore getInstance() {
```

Comment:
The static singleton pattern always makes me wary.  It can be a serious pain to build unit tests around it properly.  Might this be done using Spring instead?